framework/src/modules/chain/modules/blocks/chain.js

```diff
@@ -836,7 +836,10 @@ Chain.prototype.deleteLastBlock = function(cb) {
 			},
 			updateSystemHeaders(seriesCb) {
 				// Update our own headers: broadhash and height
-				components.system.update(seriesCb);
+				return components.system
+					.update()
+					.then(() => setImmediate(seriesCb))
+					.catch(seriesCb);
```

Comment:
What is the rationale behind the change of `update` to async? I see the code only gets more complicated. The `system.update` is only used in the surroundings of `async` library for now.

Reply:
The idea is to remove callbacks all over around while we refactor.

Reply:
I would feel more comfortable with leaving the system.update working with callbacks.

I am not convinced about the way how it's implemented now:

1. system.update is an **async** function - `async update()`
2. in both of the calls in "chain" and "loader" it's called like a **promise** - `return components.system.update().then(() => seriesCb()).catch(seriesCb);`
3. both of the functions "chain.deleteLastBlock" and "loader.sync" are using **callbacks** and async library.

Previously there were callback used consistently.

Reply:
@nazarhussain, @shuse2 and @jondubois raised the point during another PR https://github.com/LiskHQ/lisk/pull/2890. Tackling it in a separate issue was the safe strategy to avoid any behavioural change.
The idea is to not to leave any callback behind. I've followed the same pattern in another components.

Reply:
@MaciejBaj Problem is not the usage, problem is the function definition. A function which is returning a promise and accepting a callback is a code smell, so if we get chance we should remove such straight forward issues. 

- Want callback then don't return Promise 
- Want a promise, then don't accept a callback 

That's what @diego-G is trying to resolve here, using a consistent approach.